### PR TITLE
Separate lint and unit tests in CI

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Splits the `test` invoke task into two tasks:
- the existing `test` task remains. In CI, the `--skip-linters` option is passed to only run unit tests.
- a new `lint-go` task is added, to run only the linters.

All CI jobs that used to do unit tests are now split into two jobs, one to run unit tests, the other to run linters.

### Motivation

This allows to run linters as a standalone action.
This should reduce the total time spent on the unit testing stage by 3\~4 minutes in the Gitlab CI, at the cost of an increased number of parallel jobs.
This should also slightly reduce the total time spent in Github Actions CI for Windows tests: the lint and unit tests jobs, run in parallel, complete in 35\~40 minutes instead of 40\~45 minutes before.
This should also slightly reduce the total time spent in CircleCI for unit tests: the lint and unit tests jobs, run in parallel, complete in 15 minutes instead of 20 minutes before.

### Additional Notes

n/a

### Possible Drawbacks / Trade-offs

This splits the Windows unit tests scripts into two scripts, one for unit tests, the other for linting. Is this a problem for local Windows development?

### Describe how to test/QA your changes

Check the result of the new CI jobs.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by clarkb7 — APPROVED** [review]
[system-probe](https://github.com/DataDog/datadog-agent/blob/main/tasks/winbuildscripts/sysprobe.ps1#L33-L38) has its own lint call that could be split out.

**Review by bkabrda — APPROVED** [review]

**Comment by KSerrania** [comment]
Since that would require a review from another team, I'll take care of splitting the system-probe lint and test jobs in a separate PR, and merge this one as-is.

File: .github/workflows/windows-lint-go.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  windows-lint-go:
+    runs-on: windows-2019 # https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md
+    steps:
+      - name: Set git to use LF # needed so gofmt doesn't think every file needs reformatting
```
[the remark this answers is not in the document]
Comment (KSerrania) [reply .github/workflows/windows-lint-go.yml]: Done!

File: README.md [hunk 2]
```diff
@@ -79,9 +79,19 @@ on setting up a windows dev environment, refer to [Windows Dev Env](devenv).
...
 ## Testing
 
-Run tests using `invoke test`. During development, add the `--skip-linters` option to skip straight to the tests.
+Run all go linters and unit tests using `invoke test`.
 ```
-invoke test --targets=./pkg/aggregator/... --skip-linters
+invoke test --targets=./pkg/aggregator
+```
+
+You can add the `--skip-linters` option to skip go linters and only run the unit tests.
```
Comment (rtrieu) [inline README.md:87]:
```suggestion
You can add the `--skip-linters` option to skip go linters and run just the unit tests.
```

File: README.md [hunk 3]
```diff
@@ -79,9 +79,19 @@ on setting up a windows dev environment, refer to [Windows Dev Env](devenv).
...
+```
+invoke test --targets=./pkg/aggregator --skip-linters
+```
+
+You can also use `invoke lint-go` to only run the go linters.
```
Comment (rtrieu) [inline README.md:92]:
```suggestion
You can also use `invoke lint-go` to run just the go linters.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Minor suggestion but looks good.

**Review by sgnn7 — APPROVED** [review]

**Review by rtrieu — APPROVED** [review]
Looks good!

**Pull request merged**